### ISSUE TITLE
Thrift version update and unification

### ISSUE THREAD

**Issue by andrey-dubnik** [temporalio/temporal#2320]
**Is your feature request related to a problem? Please describe.**

Hi

I have a question which came out from the security scan we did recently. Is there a reason on why thrift (github.com/apache/thrift@v0.0.0-20161221203622-b2a4d4ae21c7) is pinned? It does look like this one is from version 0.10. There is another one used by the ringpop-go (actually 2) one from 0.9.3 and another one from the no longer maintained repo. Latest thrift is 0.15 and beyond 0.13 have some active CVEs. Could you please consider reviewing the thrift versions in those 2 packages?

Here is some dependency graph

```
go.temporal.io/server github.com/apache/thrift@v0.0.0-20161221203622-b2a4d4ae21c7
go.temporal.io/server github.com/temporalio/ringpop-go@v0.0.0-20211012191444-6f91b5915e95
github.com/temporalio/ringpop-go@v0.0.0-20211012191444-6f91b5915e95 github.com/apache/thrift@v0.0.0-20150905105024-5bc8b5a3a5da
github.com/temporalio/ringpop-go@v0.0.0-20211012191444-6f91b5915e95 github.com/samuel/go-thrift@v0.0.0-20191111193933-5165175b40af
```

**Describe the solution you'd like**

Minimum thrift library version of 0.13, better 0.15

**Describe alternatives you've considered**

None available

**Comment by tminusplus** [comment]
For context there has been some effort put into this outside of the Temporal team. Happy to continue pushing along my patches if this is something the Temporal team is interested in.

- PR pending on forked tchannel-go for update to v0.13.0 at https://github.com/temporalio/tchannel-go/pull/1.
- Opened an issue with uber/tchannel-go asking them to revisit updating Thrift at https://github.com/uber/tchannel-go/issues/844.
- Have a patch ready to update the forked ringpop-go to support v0.13.0 once tchannel-go is updated.

Happy to change my patches to support whatever version the Temporal team would like to use. I think that v0.13.0 is likely to be the most compatible with other Temporal dependencies, which is useful for mono-repos. That said, updating to a newer version has benefits as well. 

I will also be testing these patches on a three-node cluster in the next week or so.

**Comment by yiminc** [comment]
cc @mcbryde

**Comment by tminusplus** [comment]
I added a branch to update to v0.15.0, the latest. I will open the PR for review once I have tested the ringpop-go changes, and then tested Temporal in a production environment with it.

**Comment by tminusplus** [comment]
PR is opened https://github.com/temporalio/tchannel-go/pull/2

**Comment by tminusplus** [comment]
This should be done now with https://github.com/temporalio/temporal/pull/3250

**Comment by vavdoshka** [comment]
Flagging this issue with updated urgency: `github.com/apache/thrift` v0.21.0 and v0.22.0 are both affected by **GHSA-wf45-q9ch-q8gh** ("Apache Thrift TFramedTransport Go language implementation has an Integer Overflow or Wraparound vulnerability"), which is only fixed in **v0.23.0** (released 2026-04-27).

This means the current `main` branch (`v0.22.0`) is still vulnerable. PR #10202 (Dependabot) already targets this bump — would be great to get it reviewed and merged.
